packages/blockchain/src/index.ts

```diff
@@ -109,17 +110,17 @@ export interface BlockchainOptions {
   genesisBlock?: Block
 
   /**
-   * Optional consensus class. If not provided, this will default to the consensus
-   * class determined by the consensus algorithm set in the {@link Common} instance.
+   * Optional consensus algorithm. If not provided, this will default to the
+   * consensus algorithm set in the {@link Common} instance.
    */
-  consensus?: ConsensusBase
+  consensus?: ConsensusAlgorithm
```

Comment:
If you don't have a really strong reason I am not seeing right now there shouldn't be such an extra parameter and `consensus` should solely taken from the `Common` instance. If there is the need (for whatever reasons) to set on another consensus this should be done via `Common` as well - even if it might be slightly more complicated (by instantiating a default one and then set a new consensus).

Reply:
Removed